tests/integration/cloudformation/api/test_stack_policies.py

```diff
@@ -0,0 +1,696 @@
+
+        policy = cfn_client.get_stack_policy(StackName=stack.stack_name)
+        snapshot.match("policy", policy)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip(reason="Not implemented")
+    def test_not_json_policy(self, cfn_client, deploy_cfn_template, snapshot):
+        """Test to validate the error response when setting and Invalid Policy"""
+        stack = deploy_cfn_template(
+            template_path=os.path.join(
+                os.path.dirname(__file__), "../../templates/stack_policy_test.yaml"
+            ),
+            parameters={"TopicName": f"topic-{short_uid()}", "BucketName": f"bucket-{short_uid()}"},
+        )
+
+        with pytest.raises(botocore.exceptions.ClientError) as ex:
+            cfn_client.set_stack_policy(StackName=stack.stack_name, StackPolicyBody=short_uid())
+
+        error_response = ex.value.response["Error"]
+        snapshot.match("error", error_response)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip(reason="Not implemented")
+    def test_different_principal_attribute(self, cfn_client, deploy_cfn_template, snapshot):
+        stack = deploy_cfn_template(
+            template_path=os.path.join(
+                os.path.dirname(__file__), "../../templates/sns_topic_parameter.yml"
+            ),
+            parameters={"TopicName": f"topic-{short_uid()}"},
+        )
+
+        policy = {
+            "Statement": [
+                {
+                    "Effect": "Deny",
+                    "Action": "Update:*",
+                    "Principal": short_uid(),
+                    "Resource": "*",
+                }
+            ]
+        }
+        with pytest.raises(botocore.exceptions.ClientError) as ex:
+            cfn_client.set_stack_policy(
+                StackName=stack.stack_name, StackPolicyBody=json.dumps(policy)
+            )
+
+        error_response = ex.value.response["Error"]
+        snapshot.match("error", error_response)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip(reason="Not implemented")
+    def test_different_action_attribute(self, cfn_client, deploy_cfn_template, snapshot):
+        stack = deploy_cfn_template(
+            template_path=os.path.join(
+                os.path.dirname(__file__), "../../templates/sns_topic_parameter.yml"
+            ),
+            parameters={"TopicName": f"topic-{short_uid()}"},
+        )
+
+        policy = {
+            "Statement": [
+                {
+                    "Effect": "Deny",
+                    "Action": "Delete:*",
+                    "Principal": short_uid(),
+                    "Resource": "*",
+                }
+            ]
+        }
+        with pytest.raises(botocore.exceptions.ClientError) as ex:
+            cfn_client.set_stack_policy(
+                StackName=stack.stack_name, StackPolicyBody=json.dumps(policy)
+            )
+
+        error_response = ex.value.response["Error"]
+        snapshot.match("error", error_response)
```

Comment:
the usual pattern is `snapshot.match("...", ex.value.response)

any particular reason to deviate from it here?

Reply:
I'll change this.